This patch makes RubyGems look inside ${libdir}/rubyee/ for
Gems. This separates vanilla MRI 1.8 from REE. --a3li

--- lib/rubygems/defaults.rb.orig	2010-05-13 12:33:03.771476368 +0200
+++ lib/rubygems/defaults.rb	2010-05-13 12:32:50.395473295 +0200
@@ -1,3 +1,4 @@
+# This file is altered for Ruby Enterprise Edition on Gentoo.
 module Gem
 
   @post_install_hooks   ||= []
@@ -25,7 +26,7 @@
       File.join(ConfigMap[:libdir], ConfigMap[:ruby_install_name], 'gems',
                 ConfigMap[:ruby_version])
     else
-      File.join(ConfigMap[:libdir], ruby_engine, 'gems',
+      File.join(ConfigMap[:sitedir].gsub(/site_ruby\/?$/, ''), 'gems',
                 ConfigMap[:ruby_version])
     end
   end
@@ -34,7 +35,7 @@
   # Path for gems in the user's home directory
 
   def self.user_dir
-    File.join(Gem.user_home, '.gem', ruby_engine,
+    File.join(Gem.user_home, '.gem', 'rubyee',
               ConfigMap[:ruby_version])
   end
 
